ui/store/action-queue/index.test.js

```diff
@@ -0,0 +1,173 @@
+import sinon from 'sinon';
+
+import {
+  __TEST_CLEAR_QUEUE,
+  callBackgroundMethod,
+  submitRequestToBackground,
+  _setBackgroundConnection,
+} from '.';
+
+// This file tests only MV3 queue scenario
+// MV2 tests are already covered by '../actions.test.js'
+
+jest.mock('../../../shared/modules/mv3.utils', () => {
+  return {
+    isManifestV3: () => true,
+  };
+});
+
+describe('ActionQueue', () => {
+  afterEach(() => {
+    sinon.restore();
+    __TEST_CLEAR_QUEUE();
+  });
+  describe('submitRequestToBackground', () => {
+    it('calls promisified background method if the stream is connected', () => {
+      const background = {
+        connectionStream: {
+          readable: true,
+        },
+        backgroundFunction1: sinon.stub(),
+      };
+
+      _setBackgroundConnection(background);
+      submitRequestToBackground('backgroundFunction1');
+      expect(background.backgroundFunction1.called).toStrictEqual(true);
+    });
+
+    it('does not calls promisified background method if the stream is not connected', () => {
+      const background = {
+        connectionStream: {
+          readable: false,
+        },
+        backgroundFunction2: sinon.stub(),
+      };
+
+      _setBackgroundConnection(background);
+      submitRequestToBackground('backgroundFunction2');
+      expect(background.backgroundFunction2.called).toStrictEqual(false);
+    });
+
+    it('calls promisified background method on stream reconnection', () => {
+      // This is not an adequate simulation of what reconnection would look like
```

Comment:
Hey @naugtur : do you have a better suggestion for this ?

Reply:
It's all synchronous. I'd like a version of that where an action gets interrupted by a disconnect and the test also covers recovery and draining the queue.

Reply:
I improved it but forgot to remove the comment :D